File: processes/readingEmptyFile/systemCall.c
#include <stdio.h>
#include <sys/time.h>

int main(int argc, char **argv) {
  int rounds = 10000;
  double totalTime = 0;
  struct timeval startTime;
  struct timeval endTime;
  for (int i = 0; i < rounds; ++i) {
    gettimeofday(&startTime, NULL);
    FILE* x = fopen("./empty.txt", "w");
    fclose(x);
    gettimeofday(&endTime, NULL);
    totalTime += endTime.tv_usec - startTime.tv_usec;
  }
  printf("Average time for reading empty file: %f microseconds\n", totalTime / rounds);
}

File: processes/readingEmptyFile/systemCall.js
const fs = require('fs');


let startTime = 0, endTime = 0, totalTime = 0, rounds = 10000;
for (let i = 0; i < rounds; ++i) {
  startTime = process.hrtime();
  fs.readFileSync('./empty.txt')
  endTime = process.hrtime()
  totalTime += (endTime[1] - startTime[1]) / 1000;
}
console.log(`Average time for reading empty file: ${totalTime / rounds}  microseconds\n`);